### PR TITLE
Update domain only flow: `site-picker` step

### PR DESCRIPTION
**Pull request by gius80**
## Changes proposed in this Pull Request

This PR enables the reskinned design on domain only flow and updates the `site-picker` step.

It's part of the new domain-only flow (pcYYhz-ye-p2)

site-picker-before

site-picker-after

## Testing instructions

- Build this branch locally or open the live Calypso link
- Navigate to to `/start/domain/domain-only`
- Enter a domain name and select it, in order to move to the next step
- Select `Existing WordPress.com site` on the subsequent step (you need to be logged in to see this option)
- Verify that the layout matches the screenshots above

### CODE REVIEW

File: client/signup/config/steps-pure.js [hunk 1]
```diff
@@ -441,7 +441,7 @@ export function generateSteps( {
 			apiRequestFunction: createSiteOrDomain,
 			props: {
 				get headerText() {
-					return i18n.translate( 'Choose your site?' );
+					return i18n.translate( 'Choose your site' );
```
Comment (a8ci18n) [inline client/signup/config/steps-pure.js:444]: :warning: This change will be queued for retranslation. We'll display the English text until then.

File: client/blocks/site/index.jsx [hunk 2]
```diff
@@ -151,19 +156,26 @@ class Site extends Component {
 							: site.domain
 					}
 				>
-					<SiteIcon site={ site } size={ this.props.compact ? 24 : 32 } />
+					<SiteIcon
+						defaultIcon={ this.props.isReskinned ? layout : null }
+						site={ site }
+						// eslint-disable-next-line no-nested-ternary
+						size={ this.props.compact ? 24 : this.props.isReskinned ? 50 : 32 }
```
Comment (leonardost) [inline client/blocks/site/index.jsx:179]: I'm not particularly a fan of nested ternaries 😄 but I see there was already another one in this file and this one is pretty straightforward, so I guess it's ok.

File: client/blocks/site/index.jsx [hunk 3]
```diff
@@ -194,6 +206,18 @@ class Site extends Component {
...
+							<div className="site__domain">
+								{ /* eslint-disable-next-line no-nested-ternary */ }
+								{ this.props.isNavUnificationEnabled && ! isEnabled( 'jetpack-cloud' )
+									? site.domain
+									: this.props.homeLink
+									? translate( 'View %(domain)s', {
+											args: { domain: site.domain },
+									  } )
+									: site.domain }
+							</div>
```
Comment (leonardost) [inline client/blocks/site/index.jsx]: Looks like this code is the same as the other section above, so I think it's a good idea to refactor this into a function, something like `renderDomainSection`. As Robert Martin says (though It's cliche and overused) "duplicated code may be the root of all evil in software" 😆

**Review by leonardost — APPROVED** [review]
LGTM, this update looks really nice too! Left just one nitpicky comment there.

File: client/blocks/site/index.jsx (hunk 3 above)
Comment (gius80) [reply client/blocks/site/index.jsx]: I couldn’t agree more with you and Uncle Bob! 😁

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7120047).

**Pull request merged**